src/views/ModelPlan/TaskList/Beneficiaries/Frequency/index.tsx

```diff
@@ -362,28 +368,53 @@ const Frequency = () => {
                         scrollElement="precedenceRules"
                         error={!!flatErrors.precedenceRules}
                       >
-                        <Label
-                          htmlFor="beneficiaries-precedence-rules"
-                          className="maxw-none"
-                        >
+                        <Label htmlFor="precedenceRules">
                           {beneficiariesT('precedenceRules.label')}
                         </Label>
 
-                        <p className="text-base margin-0 line-height-body-3">
+                        <p className="text-base margin-top-1 margin-bottom-0 line-height-body-3">
                           {beneficiariesT('precedenceRules.sublabel')}
                         </p>
 
                         <FieldErrorMsg>
                           {flatErrors.precedenceRules}
                         </FieldErrorMsg>
 
-                        <Field
-                          as={TextAreaField}
-                          className="height-15"
-                          error={flatErrors.precedenceRules}
-                          id="beneficiaries-precedence-rules"
-                          data-testid="beneficiaries-precedence-rules"
-                          name="precedenceRules"
+                        {getKeys(beneficiaryPrecedenceConfig.options).map(
+                          key => (
+                            <Fragment key={key}>
+                              <Field
+                                as={CheckboxField}
+                                id={`beneficiaries-precedence-rules-${key}`}
+                                data-testid={`beneficiaries-precedence-rules-${key}`}
+                                name="precedenceRules"
+                                label={beneficiaryPrecedenceConfig.options[key]}
+                                value={key}
+                                checked={values.precedenceRules.includes(key)}
+                              />
+
+                              {values.precedenceRules?.includes(key) && (
+                                <div className="margin-left-4">
+                                  <span>
+                                    {beneficiariesT('precedenceRulesYes.label')}
```

Comment:
Ideally this would be decoupled from a single label, much like you did on like 407

```
{beneficiariesT('precedenceRules${beneficiaryPrecedenceConfig.options[key]}.label')}
```